### PR TITLE
Game

### PR DESCRIPTION
**Pull request by shakerAbuDrais**
Created the game class and added all the required methods

### CODE REVIEW

File: games/game.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+class Game < Item
+  attr_accessor :multiplayer, :last_played_date
+
+  def initialize(name, publish_date, multiplayer, last_played_date)
+    super(name, publish_date)
+    @multiplayer = multiplayer
+    @last_played_date = last_played_date
+  end
+
+  def to_s
```
Comment (cvalencia1991) [inline games/game.rb]: this method is not in the requirements, you can delete it

File: games/game.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    "Game: #{@name}, #{@publish_date}, #{@multiplayer}, #{@last_played_date}"
+  end
+
+  def can_be_archived?
+    current_date = Date.today
+    year = current_date.year - @publish_date.year
+    return true if year > 2
```
Comment (cvalencia1991) [inline games/game.rb]: you have to change this part of the code, like the requirement says it suppost to be:should return true if parent's method returns true AND if last_played_at is older than 2 years something like 

`super && year>2`

File: games/game.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+require_relative '../items'
+
+class Game < Item
```
Comment (cvalencia1991) [inline games/game.rb:3]: it suppost to be override no inherit, you can delete the Item

File: author/author.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+class Author
```
Comment (cvalencia1991) [inline author/author.rb:1]: this class should inherit the items it is supposed to be like

` class Author < items`

File: author/author.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  def initialize(first_name, last_name, id)
+    @first_name = first_name
+    @last_name = last_name
+    @id = id
+    @items = []
+  end
+
+  def add_item(item)
+    @items.push(item) unless @items.include?(item)
+    item.add_author(self)
```
Comment (cvalencia1991) [inline author/author.rb]: should add self as a property of the item object (by using the correct setter from the item object) this should be  like this:

` @item.author = self`

**Review by cvalencia1991 — CHANGES_REQUESTED** [review]
# Changes Required ♻
Hi @shakerAbuDrais 

Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! 
**_However, I did notice a few areas where some additional changes would further enhance the quality of the code._** 
But you are almost there! 

almost-there

## Highlights
There are some excellent points that you have made perfectly. Keep up the great work! 👏

- Well done for the effort
- Great job done with commits
- Correct Linters

However, there is always room for improvement.

## Required Changes

File: games/game.rb (hunk 1 above)
Comment (shakerAbuDrais) [reply games/game.rb:12]: this one was to make sure that everything is handled as a string so it will ease my work later, but i will remove it anyway

**Comment by cvalencia1991** [comment]
# Approved 🥳

Hi @shakerAbuDrais 

Great work nailing the project 💪

**_Your project is COMPLETE! There is nothing else to say other than... it's time to merge it! 🚀 Congratulations!_**

leonardo-dicaprio


## Highlights
There are some excellent points that you have made perfectly. Keep up the great work! 👏
- Great work using the right Git Flow  👍
- Commits are descriptive, good work.
- Linters are working properly, great job! ✅
- the requirements are good 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

N/A

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------

**Review by cvalencia1991 — APPROVED** [review]

**Pull request merged**